lib/jekyll/layout.rb

```diff
@@ -43,7 +45,11 @@ def initialize(site, base, name)
     #
     # Returns nothing.
     def process(name)
-      self.ext = File.extname(name)
+      self.mime_type = MIME::Types.find{|t| ".#{t.preferred_extension}" == File.extname(name)}
```

Comment:
What happens here when two MIME types have the same preferred extension? Does that happen? I bet it does somewhere...